forward_script/app.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+use Silex\Application;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Process\Process;
+use Wikimedia\ForwardScript\PageInformationCollector;
+use Wikimedia\ForwardScript\ApplicationException;
+
+require __DIR__.'/vendor/autoload.php';
+
+
+$app = new Application();
+
+// Settings
+$app['cache_dir'] = __DIR__.'/cache';
+$app['commons_api_url'] = 'https://commons.wikimedia.org/w/api.php';
+$app['wikipedia_api_url'] = 'https://de.wikipedia.org/w/api.php';
+$app['python_path'] = realpath( __DIR__ . '/../update-bot/' );
+$app['default_categories_config'] = $app['python_path'] . '/config/commonscat_mapping.json';
+$app['pageinfo_script'] = 'python -m wlmbots.pageinfo';
+$app['commons_upload_url'] = 'https://commons.wikimedia.org/wiki/Special:UploadWizard?';
+
+// Services
+$app['cache'] = $app->share( function ( $app ) {
+	return new \Doctrine\Common\Cache\FilesystemCache( $app['cache_dir'] );
+} );
+$app['commons_api'] = $app->share( function ( $app ) {
+	return new \Mediawiki\Api\MediawikiApi( $app[ 'commons_api_url' ] );
+} );
+$app['wikipedia_api'] = $app->share( function ( $app ) {
+	return new \Mediawiki\Api\MediawikiApi( $app[ 'wikipedia_api_url' ] );
+} );
+$app['campaign_validator'] = $app->share( function ( $app ) {
+	return new \Wikimedia\ForwardScript\CampaignValidator( $app[ 'commons_api' ] );
+} );
+$app['pageinfo'] = $app->share( function ( $app ) {
+	$defaultCategories = json_decode( file_get_contents( $app['default_categories_config'] ), true );
+	$process = new Process( $app['pageinfo_script'], $app['python_path'] );
+	return new PageInformationCollector( $app[ 'wikipedia_api' ], $process, $defaultCategories );
+} );
+
+// Error handling
+$app->error( function ( ApplicationException $e, $code ) {
+	$response = new Response( $e->getMessage() );
+	$response->headers->set( 'Content-Type', 'text/plain' );
+	return $response;
+} );
+
+$app->error( function ( \Symfony\Component\Process\Exception\RuntimeException $e, $code ) {
+	$response = new Response( $e->getMessage() );
+	$response->headers->set( 'Content-Type', 'text/plain' );
+	return $response;
+} );
+
+// Routes
+$app->get( '/', function() {
+	return 'WLM redirect script';
+} );
+
+$app->get( '/redirect/{pageName}/{campaign}/{id}/{lat}/{lon}',
```

Comment:
Not harmful in this case but I don't think having a nested route is preferable to using GET parameters here.